File: backend/financeiro/src/model/banco_agencia_model.py
from src import db
from src.model.banco_model import BancoModel


class BancoAgenciaModel(db.Model):
    __tablename__ = 'banco_agencia'

    id = db.Column(db.Integer, primary_key=True)
    numero = db.Column(db.String(20))
    digito = db.Column(db.String(1))
    nome = db.Column(db.String(100))
    telefone = db.Column(db.String(15))
    contato = db.Column(db.String(100))
    gerente = db.Column(db.String(100))
    observacao = db.Column(db.String(250))
    id_banco = db.Column(db.Integer, db.ForeignKey('banco.id'))

    banco_model = db.relationship('BancoModel', foreign_keys=[id_banco])

    def mapping(self, data):
        self.id = data.get('id')
        self.id_banco = data.get('idBanco')
        self.numero = data.get('numero')
        self.digito = data.get('digito')
        self.nome = data.get('nome')
        self.telefone = data.get('telefone')
        self.contato = data.get('contato')
        self.gerente = data.get('gerente')
        self.observacao = data.get('observacao')

    def serialize(self):
        return {
            'id': self.id,
            'idBanco': self.id_banco,
            'numero': self.numero,
            'digito': self.digito,
            'nome': self.nome,
            'telefone': self.telefone,
            'contato': self.contato,
            'gerente': self.gerente,
            'observacao': self.observacao,
            'bancoModel': self.banco_model.serialize() if self.banco_model else None,
        }

File: frontend/financeiro/lib/app/data/model/banco_agencia_model.dart
import 'dart:convert';
import 'package:pluto_grid/pluto_grid.dart';

import 'package:financeiro/app/data/model/model_imports.dart';

class BancoAgenciaModel {
	int? id;
	int? idBanco;
	String? numero;
	String? digito;
	String? nome;
	String? telefone;
	String? contato;
	String? gerente;
	String? observacao;
	BancoModel? bancoModel;

	BancoAgenciaModel({
		this.id,
		this.idBanco,
		this.numero,
		this.digito,
		this.nome,
		this.telefone,
		this.contato,
		this.gerente,
		this.observacao,
		this.bancoModel,
	});

	static List<String> dbColumns = <String>[
		'id',
		'numero',
		'digito',
		'nome',
		'telefone',
		'contato',
		'gerente',
		'observacao',
	];
	
	static List<String> aliasColumns = <String>[
		'Id',
		'Numero',
		'Digito',
		'Nome',
		'Telefone',
		'Contato',
		'Gerente',
		'Observacao',
	];

	BancoAgenciaModel.fromJson(Map<String, dynamic> jsonData) {
		id = jsonData['id'];
		idBanco = jsonData['idBanco'];
		numero = jsonData['numero'];
		digito = jsonData['digito'];
		nome = jsonData['nome'];
		telefone = jsonData['telefone'];
		contato = jsonData['contato'];
		gerente = jsonData['gerente'];
		observacao = jsonData['observacao'];
		bancoModel = jsonData['bancoModel'] == null ? BancoModel() : BancoModel.fromJson(jsonData['bancoModel']);
	}

	Map<String, dynamic> get toJson {
		Map<String, dynamic> jsonData = <String, dynamic>{};

		jsonData['id'] = id != 0 ? id : null;
		jsonData['idBanco'] = idBanco != 0 ? idBanco : null;
		jsonData['numero'] = numero;
		jsonData['digito'] = digito;
		jsonData['nome'] = nome;
		jsonData['telefone'] = telefone;
		jsonData['contato'] = contato;
		jsonData['gerente'] = gerente;
		jsonData['observacao'] = observacao;
		jsonData['bancoModel'] = bancoModel?.toJson;
	
		return jsonData;
	}
	
	String objectEncodeJson() {
		final jsonData = toJson;
		return json.encode(jsonData);
	}

	plutoRowToObject(PlutoRow plutoRow) {
		id = plutoRow.cells['id']?.value;
		idBanco = plutoRow.cells['idBanco']?.value;
		numero = plutoRow.cells['numero']?.value;
		digito = plutoRow.cells['digito']?.value;
		nome = plutoRow.cells['nome']?.value;
		telefone = plutoRow.cells['telefone']?.value;
		contato = plutoRow.cells['contato']?.value;
		gerente = plutoRow.cells['gerente']?.value;
		observacao = plutoRow.cells['observacao']?.value;
		bancoModel = BancoModel();
		bancoModel?.nome = plutoRow.cells['bancoModel']?.value;
	}	

	BancoAgenciaModel clone() {
		return BancoAgenciaModel(
			id: id,
			idBanco: idBanco,
			numero: numero,
			digito: digito,
			nome: nome,
			telefone: telefone,
			contato: contato,
			gerente: gerente,
			observacao: observacao,
		);			
	}

	
}